ipam/allocator.go

```diff
@@ -406,7 +406,7 @@ func (alloc *Allocator) Delete(ident string) error {
 func (alloc *Allocator) delete(ident string) error {
 	cidrs := alloc.removeAllOwned(ident)
 	if len(cidrs) == 0 {
-		return fmt.Errorf("Delete: no addresses for %s", ident)
+		return nil
```

Comment:
Maybe log this to stderr in case it is of interest to someone debugging?
In the typical case stderr is inherited from kubelet, i.e. the message will show up in the kubelet logs.

Reply:
Added debug log.

Reply:
Ah, sorry, I misread what you'd done.  If you want to swallow the error for CNI that code should be in the CNI plugin.
Unless I'm missing the reason why you'd want to do it for every caller?

Reply:
Actually this change goes into `weaver` process. Either called directly or through CNI plugin, either way to me this method need to be idempotent. Hence suppressing the error here.

Reply:
Yes, I'm missing why it needs to be idempotent for every caller.

Reply:
I have restricted the changes to CNI plugin only by ignoring the error. PTAL

Reply:
That is remarkably ugly :stuck_out_tongue_closed_eyes:  I guess we can go with it.

Did wonder about changing ipam to return 204 for the no-addresses case, but I see it currently returns 204 in every case!

Reply:
> That is remarkably ugly

Indeed. Could not think more idiomatic way.  Thought of adding a custom error type but call crosses process boundaries.